test/evp_extra_test.c

```diff
@@ -1207,6 +1250,10 @@ int setup_tests(void)
     ADD_ALL_TESTS(test_invalide_ec_char2_pub_range_decode,
                   OSSL_NELEM(ec_der_pub_keys));
 #endif
+#ifdef OPENSSL_NO_FIPS
```

Comment:
However, you can always do the same I did for `provider_internal_test` here, in `build.info`:

```
    IF[{- $disabled{fips} -}]
      DEFINE[evp_extra_test]=NO_FIPS_MODULE
    ENDIF
```

and use that as a guard...

Reply:
It seems strange to me to deliberately suppress the OPENSSL_NO_FIPS macro - only to then create a new one instead.

Reply:
The difference is that this is telling this specific test app if the fips module is available or not, i.e. entirely based on what's being built here and now, rather than on something that looks like a library feature.

Reply:
BTW, the condition should be this:

```
    IF[{- $disabled{fips} || !$target{dso_scheme} -}]
```

The lack of a DSO scheme doesn't necessarily mean that the module isn't built, it only means that we can't load it.

Reply:
Speaking of macros, this had me rethink that macro name I used for the provider tests.  See #8664

Reply:
I thought we'd decided on `FIPS_MODE` as the macro to be defined.

Reply:
You may want to read https://www.openssl.org/docs/OpenSSL300Design.html#code-maintenance again.  `FIPS_MODE` isn't meant to be used to say if the fips provider is available or not.